framework/grid/GridView.php

```diff
@@ -503,7 +503,7 @@ public function renderTableRow($model, $key, $index)
         foreach ($this->columns as $column) {
             $cells[] = $column->renderDataCell($model, $key, $index);
         }
-        if ($this->rowOptions instanceof Closure) {
+        if (is_callable($this->rowOptions)) {
```

Comment:
please look at https://github.com/yiisoft/yii2/issues/12711 for an integral approach